src/main/java/org/opentripplanner/index/IndexAPI.java

```diff
@@ -630,6 +632,7 @@ public Response getGraphQL (String query, @HeaderParam("OTPTimeout") @DefaultVal
     @Path("/graphql/batch")
     @Consumes(MediaType.APPLICATION_JSON)
     public Response getGraphQLBatch (List<HashMap<String, Object>> queries, @HeaderParam("OTPTimeout") @DefaultValue("10000") int timeout, @HeaderParam("OTPMaxResolves") @DefaultValue("1000000") long maxResolves) {
+        int final_timeout = ChecktimeOut(timeout);
```

Comment:
Timeout should only be altered if NOT specified in the http header (only modify timeouts for requests NOT specifying the OTPTimeout header)